runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientProduceFactory.java

```diff
@@ -1318,8 +1318,7 @@ private void onClientInitialFlush(
 
             assert initialAck <= initialSeq;
 
-            final int noAck = (int) (initialSeq - initialAck);
-            doClientInitialWindow(traceId, noAck, noAck + initialBudgetMax);
+            doClientInitialWindow(traceId, 0, initialBudgetMax);
```

Comment:
Are you sure this is correct?
Would expect the reaction to `DATA` and `FLUSH` to be the same with regards to flow control, given they can both advance the `sequence` and indicate `reserved`.

Reply:
If we check the previous code, noAck = initialSeq - initialAck.
So in the doClienttInitialWindow: 
`final long newInitialAck = Math.max(initialSeq - minInitialNoAck, initialAck);`
But in reality this is:
`final long newInitialAck = Math.max(initialAck, initialAck);`
initialAck is 0 in the beginning, so basically it'll never be increased, since newInitialAck > initialAck condition will never be true. I think when we're doing the data path we're not increasing the initialAck in the doClientInitialWindow call which we're doing from the onClientInitialData, but when the actual window frame is coming back from the network.